src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -114,8 +114,8 @@ public class BlockStoreQueue : IBlockStore, IDisposable
 
             if (this.storeSettings.ReIndex)
             {
-                this.logger.LogTrace("(-)[REINDEX_NOT_SUPPORTED]");
-                throw new NotSupportedException("Re-indexing the block store in currently not supported.");
+                await this.blockRepository.SetTxIndexAsync(this.storeSettings.TxIndex).ConfigureAwait(false);
+                await this.blockRepository.ReIndexAsync().ConfigureAwait(false);
```

Comment:
Should TxIndex perhaps be set back after the ReIndex? (Not sure, just wondering)

Reply:
The expected functionality is that we're setting the flag indefinitely. So we either always index transactions, or always don't. So the flag should stay set :)